### PR TITLE
chore: remove custom searcher and DSAT

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

removes custom searcher and DSAT

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/get-started/example-solutions/_index.rst [hunk 1]
```diff
@@ -55,29 +55,6 @@ For an introduction to using the training APIs, please visit :ref:`Training APIs
...
-      -  ImageNet (Generated)
-      -  :download:`torchvision.tgz </examples/torchvision.tgz>`
-
-   -  -  Hugging Face (DeepSpeed/PyTorch)
-      -  Beans (Hugging Face)
-      -  :download:`hf_image_classification.tgz </examples/hf_image_classification.tgz>`
-
-   -  -  Hugging Face (DeepSpeed/PyTorch)
-      -  WikiText (Hugging Face)
-      -  :download:`hf_language_modeling.tgz </examples/hf_language_modeling.tgz>`
```
Comment (rb-determined-ai) [inline docs/get-started/example-solutions/_index.rst:79]: I don't think these should be deleted, but they should be recategorized as plain DeepSpeed examples.  Or maybe the last two could be classified as HF examples.

However, within these examples, there are dsat details that need stripping out.  (I forwarded those to you in slack)

File: master/pkg/searcher/operations.go [hunk 2]
```diff
@@ -32,12 +32,11 @@ type (
 
 // All the operation types that support serialization.
 const (
-	CreateOperation              OperationType = 0
-	TrainOperation               OperationType = 1
-	ValidateOperation            OperationType = 2
-	CloseOperation               OperationType = 4
-	ValidateAfterOperation       OperationType = 5
-	SetSearcherProgressOperation OperationType = 6
```
Comment (rb-determined-ai) [inline master/pkg/searcher/operations.go:40]: this change reminds me...

Have you checked the case where a custom-search experiment was in progress before the upgrade?  The experiment can fail in an ugly way, just as long as the master doesn't panic or anything.

**Review by rb-determined-ai — COMMENTED** [review]
very nice.  -11k!

**Comment by jgongd** [comment]
For my understanding, is deep speed also part of the searcher context as well?

**Review by jgongd — APPROVED** [review]
Nice work!

File: master/pkg/searcher/custom_search.go [hunk 3]
```diff
@@ -1,147 +0,0 @@
...
-		CustomSearchProgress float64
-	}
-
-	customSearch struct {
-		expconf.CustomConfig
-		customSearchState
-	}
-)
-
-func newCustomSearch(config expconf.CustomConfig) SearchMethod {
```
Comment (amandavialva01) [inline master/pkg/searcher/custom_search.go:27]: I believe the `expconf.CustomConfig` type still exists, do we want to remove this as well?

Comment (amandavialva01) [reply master/pkg/searcher/custom_search.go:27]: This type only seems to be used [here](https://github.com/determined-ai/determined/blob/remove-custom-searcher/master/internal/db/postgres_experiments_intg_test.go#L759) in our experiment integration tests

**Review by amandavialva01 — COMMENTED** [review]
Backend looks great! Left a few comments on a couple of structs that can potentially be removed

File: docs/get-started/example-solutions/_index.rst (hunk 1 above)
Comment (azhou-determined) [reply docs/get-started/example-solutions/_index.rst]: done

File: master/pkg/searcher/operations.go (hunk 2 above)
Comment (azhou-determined) [reply master/pkg/searcher/operations.go:40]: good point, i added back the configs in accordance with how other searchers were removed. now, restarting a custom search experiment will just log an error.

File: master/pkg/searcher/custom_search.go (hunk 3 above)
Comment (azhou-determined) [reply master/pkg/searcher/custom_search.go:27]: good point, removed it and the calling piece in that intg test

File: master/pkg/searcher/search_method.go [hunk 4]
```diff
@@ -85,8 +76,6 @@ func NewSearchMethod(c expconf.SearcherConfig) SearchMethod {
 		return newAsyncHalvingSearch(*c.RawAsyncHalvingConfig, c.SmallerIsBetter())
 	case c.RawAdaptiveASHAConfig != nil:
 		return newAdaptiveASHASearch(*c.RawAdaptiveASHAConfig, c.SmallerIsBetter())
-	case c.RawCustomConfig != nil:
-		return newCustomSearch(*c.RawCustomConfig)
```
[the remark this answers is not in the document]
Comment (azhou-determined) [reply master/pkg/searcher/search_method.go:89]: good idea, but apparently we can't 'cause of the case where master tries to restore a pre-upgrade custom search experiment. so i've kept the config but treat it like how we treat the other removed searcher configs.

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
There's an import of `ProjectSpec` from the deleted file `webui/react/src/services/stream/wire.ts` in `webui/react/src/stores/projects.tsx`

**Comment by azhou-determined** [comment]
> There's an import of `ProjectSpec` from the deleted file `webui/react/src/services/stream/wire.ts` in `webui/react/src/stores/projects.tsx`

sorry, i'm on a new laptop and having some proto/bindings build issues. those files were added back 😄

**Review by rb-determined-ai — APPROVED** [review]
siiiick

**Pull request merged**